sympy/stats/random_matrix_models.py

```diff
@@ -207,11 +209,49 @@ def joint_eigen_distribution(mat):
     >>> from sympy.stats import GaussianUnitaryEnsemble as GUE
     >>> from sympy.stats import joint_eigen_distribution
     >>> U = GUE('U', 2)
-    >>> joint_eigen_dsitribution(U)
+    >>> joint_eigen_distribution(U)
     Lambda((l[1], l[2]), exp(-l[1]**2 - l[2]**2)*Product(Abs(l[_i] - l[_j])**2, (_j, _i + 1, 2), (_i, 1, 1))/pi)
     """
+    if not isinstance(mat, RandomMatrixSymbol):
+        raise ValueError("%s is not of type, RandomMatrixSymbol.")
     return mat.pspace.model.joint_eigen_distribution()
 
+def JointEigenDistribution(mat):
+    """
+    Creates joint distribution of eigen values of matrices with random
+    expressions.
+
+    Parameters
+    ==========
+
+    mat: Matrix
+        The matrix under consideration
+
+    Returns
+    =======
+
+    JointDistributionHandmade
+
+    Examples
+    ========
+
+    >>> from sympy.stats import Normal, JointEigenDistribution
+    >>> from sympy import Matrix
+    >>> A = [[Normal('A00', 0, 1), Normal('A01', 0, 1)],
+    ... [Normal('A10', 0, 1), Normal('A11', 0, 1)]]
+    >>> JointEigenDistribution(Matrix(A))
+    JointDistributionHandmade(-sqrt(A00**2 - 2*A00*A11 + 4*A01*A10 + A11**2)/2
+    + A00/2 + A11/2, sqrt(A00**2 - 2*A00*A11 + 4*A01*A10 + A11**2)/2 + A00/2 + A11/2)
+
+    """
+    eigen_dict = mat.eigenvals()
+    eigenvals = eigen_dict.keys()
+    sorted(eigenvals, key=lambda val: eigen_dict[val])
```

Comment:
I have tried `eigenvals` canonical by using `sorted`. @sidhantnagpal will the key(multiplicity of the eigen value ) always work?

Reply:
You can get the list (with multiplicity) using:
```py
evs = mat.eigenvals(multiple=True)
```
and return the result like `JointDistributionHandmade(*evs)`.

Reply:
Will it always be the same or the order will be random?

Reply:
`eigenvals` internally calls [`roots`](https://docs.sympy.org/latest/modules/polys/reference.html#sympy.polys.polyroots.roots) which will make same values appear next to each other.